lib/serviceClient.ts

```diff
@@ -185,9 +199,26 @@ export class ServiceClient {
       options = {};
     }
 
+    if (options.baseUri) {
+      this.baseUri = options.baseUri;
+    }
+
     let serviceClientCredentials: ServiceClientCredentials | undefined;
     if (isTokenCredential(credentials)) {
-      serviceClientCredentials = new AzureIdentityCredentialAdapter(credentials);
+      let scope: string | undefined = undefined;
+      const azureManagementClouds = [
```

Comment:
Nit: This const can be defined outside of this class to avoid initialization every time a ServiceClient is created

Reply:
Addressed here: https://github.com/Azure/ms-rest-js/pull/457/commits/7224651161fbf7928a08f861a864d5dd34064ac1